### PR TITLE
gh-84436: update docs on Py_None/Py_True/Py_False/Py_Ellipsis becoming immortal

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-84436

----
:books: Documentation preview :books:: https://cpython-previews--105195.org.readthedocs.build/

### CODE REVIEW

**Review by erlend-aasland — APPROVED** [review]

File: Doc/c-api/bool.rst [hunk 1]
```diff
@@ -19,29 +19,30 @@ are available, however.
 
 .. c:var:: PyObject* Py_False
 
-   The Python ``False`` object.  This object has no methods.  It needs to be
-   treated just like any other object with respect to reference counts.
+   The Python ``False`` object.  This object has no methods and is immortal.
```
Comment (JelleZijlstra) [inline Doc/c-api/bool.rst]: Could this link to a glossary entry or docs section on immortal objects? Users may have no idea what this means.

File: Doc/c-api/bool.rst [hunk 2]
```diff
@@ -19,29 +19,30 @@ are available, however.
...
+
+.. versionchanged:: 3.12
+   :const:`Py_False` is immortal.
```
Comment (JelleZijlstra) [inline Doc/c-api/bool.rst]: `:c:const:` maybe?

**Review by JelleZijlstra — DISMISSED** [review]

File: Doc/c-api/bool.rst (hunk 2 above)
Comment (iritkatriel) [reply Doc/c-api/bool.rst]: I think I picked up ``:const:`` from the Py_RETURN_FALSE doc. 

Is there such a thing as ``:c:const:`` (grep on the docs dir doesn't give me any).

File: Doc/c-api/bool.rst (hunk 1 above)
Comment (iritkatriel) [reply Doc/c-api/bool.rst]: I don't think we have anything in the docs. I can link to the PEP.

File: Doc/c-api/bool.rst (hunk 2 above)
Comment (JelleZijlstra) [reply Doc/c-api/bool.rst]: I think we actually want ":c:data:". That's how the link to Py_None in https://docs.python.org/3.10/c-api/slice.html#c.Py_Ellipsis is created.

File: Doc/c-api/bool.rst (hunk 1 above)
Comment (JelleZijlstra) [reply Doc/c-api/bool.rst]: Ideally there should be a new section in the docs, but linking to the PEP is fine for now.

File: Doc/c-api/bool.rst (hunk 2 above)
Comment (iritkatriel) [reply Doc/c-api/bool.rst]: I had :c:data: initially but there was some build error. I'll try it again.

File: Doc/c-api/bool.rst [hunk 3]
```diff
@@ -19,29 +19,32 @@ are available, however.
 
 .. c:var:: PyObject* Py_False
 
-   The Python ``False`` object.  This object has no methods.  It needs to be
-   treated just like any other object with respect to reference counts.
+   The Python ``False`` object.  This object has no methods and is
+   `immortal.<https://peps.python.org/pep-0683/>`_
```
Comment (iritkatriel) [inline Doc/c-api/bool.rst]:
```suggestion
   `immortal <https://peps.python.org/pep-0683/>`_.
```

File: Doc/c-api/bool.rst [hunk 4]
```diff
@@ -19,29 +19,32 @@ are available, however.
...
+.. versionchanged:: 3.12
+   :c:data:`Py_False` is immortal.
 
 
 .. c:var:: PyObject* Py_True
 
-   The Python ``True`` object.  This object has no methods.  It needs to be treated
-   just like any other object with respect to reference counts.
+   The Python ``True`` object.  This object has no methods and is
+   `immortal.<https://peps.python.org/pep-0683/>`_
```
Comment (iritkatriel) [inline Doc/c-api/bool.rst]:
```suggestion
   `immortal <https://peps.python.org/pep-0683/>`_.
```

File: Doc/c-api/none.rst [hunk 5]
```diff
@@ -15,12 +15,12 @@ same reason.
 
 .. c:var:: PyObject* Py_None
 
-   The Python ``None`` object, denoting lack of value.  This object has no methods.
-   It needs to be treated just like any other object with respect to reference
-   counts.
+   The Python ``None`` object, denoting lack of value.  This object has no methods
+   and is `immortal.<https://peps.python.org/pep-0683/>`_.
```
Comment (iritkatriel) [inline Doc/c-api/none.rst]:
```suggestion
   and is `immortal <https://peps.python.org/pep-0683/>`_.
```

File: Doc/c-api/slice.rst [hunk 6]
```diff
@@ -118,6 +118,9 @@ Ellipsis Object
 
 .. c:var:: PyObject *Py_Ellipsis
 
-   The Python ``Ellipsis`` object.  This object has no methods.  It needs to be
-   treated just like any other object with respect to reference counts.  Like
-   :c:data:`Py_None` it is a singleton object.
+   The Python ``Ellipsis`` object.  This object has no methods.  Like
+   :c:data:`Py_None` it is an `immortal<https://peps.python.org/pep-0683/>`_
```
Comment (iritkatriel) [inline Doc/c-api/slice.rst]:
```suggestion
   :c:data:`Py_None` it is an `immortal <https://peps.python.org/pep-0683/>`_.
```

File: Doc/c-api/bool.rst [hunk 7]
```diff
@@ -19,29 +19,32 @@ are available, however.
...
+
+.. versionchanged:: 3.12
+   :c:data:`Py_True` is immortal.
 
 
 .. c:macro:: Py_RETURN_FALSE
 
-   Return :const:`Py_False` from a function, properly incrementing its reference
-   count.
+   Return :const:`Py_False` from a function.
```
Comment (carljm) [inline Doc/c-api/bool.rst]:
```suggestion
   Return :c:data:`Py_False` from a function.
```

File: Doc/c-api/bool.rst [hunk 8]
```diff
@@ -19,29 +19,32 @@ are available, however.
...
 
 
 .. c:macro:: Py_RETURN_TRUE
 
-   Return :const:`Py_True` from a function, properly incrementing its reference
-   count.
+   Return :const:`Py_True` from a function.
```
Comment (carljm) [inline Doc/c-api/bool.rst]:
```suggestion
   Return :c:data:`Py_True` from a function.
```

File: Doc/c-api/bool.rst [hunk 9]
```diff
@@ -19,29 +19,32 @@ are available, however.
...
 
 
 .. c:function:: PyObject* PyBool_FromLong(long v)
 
-   Return a new reference to :const:`Py_True` or :const:`Py_False` depending on the
-   truth value of *v*.
+   Return :const:`Py_True` or :const:`Py_False`, depending on the truth value of *v*.
```
Comment (carljm) [inline Doc/c-api/bool.rst]:
```suggestion
   Return :c:data:`Py_True` or :c:data:`Py_False`, depending on the truth value of *v*.
```

File: Doc/c-api/slice.rst [hunk 10]
```diff
@@ -118,6 +118,9 @@ Ellipsis Object
 
 .. c:var:: PyObject *Py_Ellipsis
 
-   The Python ``Ellipsis`` object.  This object has no methods.  It needs to be
-   treated just like any other object with respect to reference counts.  Like
-   :c:data:`Py_None` it is a singleton object.
+   The Python ``Ellipsis`` object.  This object has no methods.  Like
+   :c:data:`Py_None` it is an `immortal <https://peps.python.org/pep-0683/>`_.
```
Comment (carljm) [inline Doc/c-api/slice.rst]:
```suggestion
   :c:data:`Py_None`, it is an `immortal <https://peps.python.org/pep-0683/>`_.
```

**Review by carljm — COMMENTED** [review]
Looks good, just a few more broken references to fix.

Since `bool.rst` is such a small file I'd be tempted to also just fix in passing the two erroring references to `Py_True` and `Py_False` near the top of the file (line 9), but that's up to you; that line isn't changed in the PR otherwise.

**Review by carljm — APPROVED** [review]

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**